packages/batch/package.json

```diff
@@ -13,25 +13,50 @@
     "test": "npm run test:unit",
     "test:unit": "jest --group=unit --detectOpenHandles --coverage --verbose",
     "jest": "jest --detectOpenHandles --verbose",
-    "test:e2e:nodejs14x": "echo 'Not Implemented'",
     "test:e2e:nodejs16x": "echo 'Not Implemented'",
     "test:e2e:nodejs18x": "echo 'Not Implemented'",
     "test:e2e:nodejs20x": "echo 'Not Implemented'",
     "test:e2e": "echo 'Not Implemented'",
     "watch": "jest --watch",
-    "build": "tsc --build --force",
+    "build:cjs": "tsc --build tsconfig.json && echo '{ \"type\": \"commonjs\" }' > lib/cjs/package.json",
+    "build:esm": "tsc --build tsconfig.esm.json && echo '{ \"type\": \"module\" }' > lib/esm/package.json",
+    "build": "npm run build:esm & npm run build:cjs",
```

Comment:
While before we were building a single output under `lib`, now we are generating two of them: one for CJS and one for ESM, respectively under `lib/cjs/*` and `lib/esm/*`.

To do this, we are now maintaining two `tsconfig.json` files, one for CJS and another for ESM, and using them to create their respective build.

In addition to this, we are also dynamically creating a `package.json` file under each one of the versions. This file contains a single section that sets the `type` to either `commonjs` or `module` (ESM). This is what allows Node.js, TS, and bundlers to recognize the folder as a module root.